test/test_torch.py

```diff
@@ -1223,6 +1223,11 @@ def test_new_tensor(self):
             self.assertEqual(res2, expected, exact_dtype=False)
             self.assertIs(torch.int, res2.dtype)
 
+            # test unpack large number
+            res3 = torch.tensor([0.1, 999999999999999999999])
```

Comment:
Let's do something like this:

```
if TEST_NUMPY:
  vals = (2**24 + 1, 2**53 + 1,  np.iinfo(np.int64).max, np.iinfo(np.uint64).max, np.iinfo(np.uint64).max + 1, -1e500, 
          1e500)
  dtypes = (torch.float32, torch.float64)
  for val, dtype in product(vals, dtypes):
    t = torch.tensor(val, dtype=dtype)
    a = np.array(val, dtype=torch_to_numpy_dtype_dict[dtype])
    self.assertEqual(t, torch.from_numpy(a))
```

Ideally this test would be refactored to be device generic so we could test it on CPU and CUDA easily, too.

Reply:
Thanks for the tip, I will implement a device generic test method based on the provided code snippet.